nat/src/main/java/org/hyperledger/besu/nat/core/AbstractNatManager.java

```diff
@@ -0,0 +1,127 @@
+/*
```

Comment:
Is it possible to implement these default implementations in the Interface or is the Abstract class needed for some reason I can't see?

Reply:
I preferred to use an abstract class because in my opinion the implementation is too big to be in an interface.

Reply:
right now I have three abstract methods in this class. for example, the doStart method is called by the start method that makes generic checks. I think it would be interesting to keep these abstract methods as well as the visibility of the fields of the class in protected. to remove 1 level of indirection I suggest removing implements in subclasses and keeping only extends keyword. Thoughts?

Reply:
That works for me!